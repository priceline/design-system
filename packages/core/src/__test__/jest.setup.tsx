import '@testing-library/jest-dom'
import 'jest-styled-components'

global.requestAnimationFrame =
  global.requestAnimationFrame ||
  function _raf(cb) {
    return setTimeout(cb, 0)
  }
