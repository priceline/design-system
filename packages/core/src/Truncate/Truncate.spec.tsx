import { render, screen } from '../__test__/testing-library'
import { Truncate } from './Truncate'

describe('Truncate', () => {
  test('renders', () => {
    const { asFragment } = render(<Truncate>renders</Truncate>)
    expect(asFragment()).toMatchSnapshot()
  })

  test('generates styles', () => {
    render(<Truncate data-testid='truncate'>generates styles</Truncate>)
    const element = screen.getByTestId('truncate')
    expect(element).toHaveStyleRule('overflow', 'hidden')
    expect(element).toHaveStyleRule('white-space', 'nowrap')
    expect(element).toHaveStyleRule('text-overflow', 'ellipsis')
  })
})
