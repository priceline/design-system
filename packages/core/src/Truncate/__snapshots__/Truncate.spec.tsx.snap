// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Truncate renders 1`] = `
<DocumentFragment>
  .c0 {
  font-family: 'Montserrat','Helvetica Neue',Helvetica,Arial,sans-serif;
  line-height: 1.4;
  font-weight: 500;
}

.c0 * {
  box-sizing: border-box;
}

.c1 {
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
}

<div
    class="c0"
  >
    <div
      class="c1"
    >
      renders
    </div>
  </div>
</DocumentFragment>
`;
